distribution/pull_v2.go

```diff
@@ -32,7 +32,11 @@ import (
 	"golang.org/x/net/context"
 )
 
-var errRootFSMismatch = errors.New("layers from manifest don't match image configuration")
+var (
+	errRootFSMismatch  = errors.New("layers from manifest don't match image configuration")
+	errMediaTypePlugin = errors.New("target is a plugin. Use `docker plugin install` to install plugins")
```

Comment:
This would be returned by the API. It may be better to return `target is a plugin`, and have the CLI recognize this error and add the `docker plugin install` suggestion. We're doing this for `docker swarm init`: https://github.com/docker/docker/blob/v1.12.0/api/client/swarm/init.go#L67